app/controllers/RegionController.scala

```diff
@@ -49,12 +49,11 @@ class RegionController @Inject() (implicit val env: Environment[User, SessionAut
 
               case None =>
                 // Assigns a region to the user on request from a signed in user
-                UserCurrentRegionTable.assignNextRegion(user.userId)
+                UserCurrentRegionTable.assignNextRegion(user.userId).get.regionId
```

Comment:
Will this function be ever called? I want to discuss about the places where this is invoked from the front-end (i.e. checking if tasks are available and whether missions are available). I would also like to discuss these checks on the server-side - whether they are relevant or not anymore in lieu of all the latest changes.

Let's discuss this once before merging this PR.